mysteriousorganism.js

```diff
@@ -0,0 +1,136 @@
+// Returns a random DNA base
+const returnRandBase = () => {
+	const dnaBases = ['A', 'T', 'C', 'G'];
+	return dnaBases[Math.floor(Math.random() * 4)];
+};
+
+// Returns a random single stand of DNA containing 15 bases
+const mockUpStrand = () => {
+	const newStrand = [];
+	for (let i = 0; i < 15; i++) {
+		newStrand.push(returnRandBase());
+	}
+	return newStrand;
+};
+
+// Create function factory
+const pAequorFactory = (specimenNum, dna) => {
+	return {
+		specimenNum,
+		dna,
+
+		mutate() {
+			const dnaBases = ['A', 'T', 'C', 'G']; // You'd probably want this variable to be global at some point.
+			const randomBase = Math.floor(Math.random() * 15);
+			const replaceBase = dna[randomBase];
+			let newBase = '';
+
+			// Find value at random place in base and replace it with a different base
+			do {
+				newBase = dnaBases[Math.floor(Math.random() * 3)];
+			} while (replaceBase === newBase);
+			dna.splice(randomBase, 1, newBase);
+			return dna;
+		},
+
+		compareDNA(inputObject) {
+			let compareCounter = 0;
+			for (i = 0; i < dna.length; i++) {
+				if (dna[i] === inputObject.dna[i]) {
+					compareCounter += 1;
+				}
+			}
+			return Math.floor((compareCounter / dna.length) * 100);
+		},
+
+		willLikelySurvive() {
+			let surviveCounter = 0;
+			for (i = 0; i < dna.length; i++) {
+				if (dna[i] === 'C' || dna[i] === 'G') {
+					surviveCounter += 1;
+				}
+			}
+			if (surviveCounter >= 9) {
```

Comment:
Je neemt hier fixed 9. Voor deze opdracht natuurlijk prima, maar wat als er ineens meer bases zijn? Dan moet je deze code herschrijven. In de compareDNA method pak je wel netjes de daadwerkelijke lengte van de bases.

Reply:
Fixed in 41db3beada8194a5fcb1ddd9390e6927fea378fa